perceval/algorithm/tomography/tomography.py

```diff
@@ -107,81 +106,65 @@ class StateTomography(AAlgorithm):
     - Computes parameters required to do state tomography
     - Performs Tomography experiment - Computes and Returns density matrices for each input state
     """
-    def __init__(self, nqubit: int, operator_processor: Processor, post_process=False, renormalization=None):
+    def __init__(self, operator_processor: Processor):
         super().__init__(processor=operator_processor)
-        self._nqubit = nqubit
         self._operator_processor = operator_processor  # Gate operation under study
-        self._backend = operator_processor.backend  # default - SLOSBackend()
-        self._source = operator_processor.source  # default - ideal source
-        self._post_process = post_process
-        self._renormalization = renormalization
-        self._heralded_modes = [(key, value) for key, value in operator_processor.heralds.items()]
-        self._size_hilbert = 2 ** nqubit
-        # Todo: I need to put this number somewhere, i forgot. find [number of elements = between (0 -> 4**nqubits-1)]
-
-    def _input_state_dist_config(self):
-        # Configures the input state for the Processor
-        input_state = BasicState("|1,0>")
-        for _ in range(1, self._nqubit):
-            # setting the input state for the gate qubit modes
-            input_state *= BasicState("|1,0>")
-        for m in self._heralded_modes:
-            # setting the input for heralded modes of the given processor
-            input_state *= BasicState([m[1]])
-        input_distribution = self._source.generate_distribution(expected_input=input_state)
-        return input_distribution
+        self._nqubit = int(operator_processor.m / 2)  # nqubit
+        self._size_hilbert = 2 ** self._nqubit
+        self._gate_logical_perf = None
 
-    def _compute_probs(self, prep_state_indices, meas_pauli_basis_indices):
+    def _configure_processor(self, prep_state_indices, meas_pauli_basis_indices):
         # Adds preparation and measurement circuit to input processor (with the gate operation under study)
         # and computes the output probability distribution.
         # param prep_state_indices: List of "nqubit" indices selecting the circuit at each qubit for a preparation state
         # param meas_pauli_basis_indices: List of "nqubit" indices selecting the circuit at each qubit for a measurement
         # circuit
-        # return: Output state probability distribution
-
-        pc = StatePreparationCircuit(prep_state_indices, self._nqubit)  # state preparation circuit object
-        mc = MeasurementCircuit(meas_pauli_basis_indices, self._nqubit)  # measurement basis circuit object
+        # returns the processor configured to perform state tomography experiment
 
-        p = Processor(self._backend, self._nqubit*2, self._source)
-        # A Processor with identical backend and source as the input
+        p = Processor(self._operator_processor.backend, self._nqubit * 2, self._operator_processor.source)
```

Comment:
Maybe we should write somewhere that tomography will use the same backend as the processor ? And do we want to do that ? If the user uses Naive it will be a very long tomography no ?

Reply:
That's how Perceval works : using the stuff inside a processor in order to make a computation. So yes, we want to do that, and no need to inform the user we're doing so, I guess.
If a user wants to simulate with Naive, well... I just hope he's patient :)

Reply:
I mentioned the default config settings of Processor within `__init__()` of Tomography